hardhat.config.ts

```diff
@@ -171,6 +177,9 @@ const config: HardhatUserConfig = {
   typechain: {
     outDir: "types",
     target: "ethers-v5",
+    alwaysGenerateOverloads: false,
```

Comment:
I think we need to put `true` here, because we have a lot of function overloading in the smart contracts.

```suggestion
    alwaysGenerateOverloads: true,
```

https://www.npmjs.com/package/typechain